people-and-planet-ai/conftest.py

```diff
@@ -0,0 +1,278 @@
+    project = os.environ["GOOGLE_CLOUD_PROJECT"]
+    run_cmd("gcloud", "config", "set", "project", project)
+
+    # Since everything requires the project, let's confiugre and show some
+    # debugging information here.
+    run_cmd("gcloud", "version")
+    run_cmd("gcloud", "config", "list")
+    return project
+
+
+@pytest.fixture(scope="session")
+def location() -> str:
+    # Override for local testing.
+    return os.environ.get("GOOGLE_CLOUD_LOCATION", "us-central1")
+
+
+@pytest.fixture(scope="session")
+def python_version() -> str:
+    return "".join(platform.python_version_tuple()[0:2])
+
+
+@pytest.fixture(scope="session")
+def unique_id() -> str:
+    id = uuid.uuid4().hex[0:6]
+    print(f"Test unique identifier: {id}")
+    return id
+
+
+@pytest.fixture(scope="session")
+def unique_name(test_name: str, unique_id: str) -> str:
+    return f"{test_name.replace('/', '-')}-{unique_id}"
+
+
+@pytest.fixture(scope="session")
+def bucket_name(test_name: str, location: str, unique_id: str) -> str:
+    # Override for local testing.
+    if "GOOGLE_CLOUD_BUCKET" in os.environ:
+        bucket_name = os.environ["GOOGLE_CLOUD_BUCKET"]
+        print(f"bucket_name: {bucket_name} (from GOOGLE_CLOUD_BUCKET)")
+        yield bucket_name
+        return
+
+    storage_client = storage.Client()
+    bucket_name = f"{test_name.replace('/', '-')}-{unique_id}"
+    bucket = storage_client.create_bucket(bucket_name, location=location)
+
+    print(f"bucket_name: {bucket_name}")
+    yield bucket_name
+
+    # Try to remove all files before deleting the bucket.
+    # Deleting a bucket with too many files results in an error.
+    try:
+        subprocess.check_call(["gsutil", "-m", "rm", "-rf", f"gs://{bucket_name}/*"])
+    except Exception:
+        # If no files were found and it fails, ignore the error.
+        pass
+
+    # Delete the bucket.
+    bucket.delete(force=True)
+
+
+@pytest.fixture(scope="session")
+# Disable printing to not log the identity token.
+@mock.patch("builtins.print", lambda x: x)
+def identity_token() -> str:
+    return (
+        run_cmd("gcloud", "auth", "print-identity-token").stdout.decode("utf-8").strip()
+    )
+
+
+def env_var(prefix: str, id: str = "") -> str:
+    return f"{prefix}_{id}".replace(".", "").replace("/", "").strip("_")
+
+
+def run_cmd(*cmd: str) -> subprocess.CompletedProcess:
+    try:
+        # TODo: add timing to see how long it took to run
+        print(f">> {cmd}")
+        p = subprocess.run(
+            cmd,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        print(p.stderr.decode("utf-8"))
+        print(p.stdout.decode("utf-8"))
+        return p
+    except subprocess.CalledProcessError as e:
+        # Include the error message from the failed command.
+        print(e.stderr.decode("utf-8"))
+        print(e.stdout.decode("utf-8"))
+        raise Exception(f"{e}\n\n{e.stderr.decode('utf-8')}") from e
```

Comment:
It's failing due to a failed shell command so it could be anywhere. I'm just doing the decoding to convert bytes to strings. I'll change it to a `RuntimeError` since we don't know what kind of command had the error.